### PR TITLE
Improve grouping to enable ogip dataset fitting 

### PR DESCRIPTION
**Pull request by luca-giunti**
@registerrier I applied some modifications to enable fitting with the `StandardOGIPDatasetDataset`.

It works (on gammapy 0.19), but I am not sure that it is the a very elegant approach. I basically override all fitting-related methods and properties to redirect to the grouped version of the dataset, which is now a simple `SpectrumDatasetOnOff`.

Before adding tests, I would like to discuss together (here or on Zoom) whether this is OK or we should consider other options.

I also added an example nb to illustrate the fit.  

Cheers :)

### CODE REVIEW

File: ogip_spectra/ogip_spectrum_dataset.py [hunk 1]
```diff
@@ -42,44 +43,127 @@ class StandardOGIPDataset(SpectrumDatasetOnOff):
...
-        """Return grouped dataset."""
-        return self._grouped_dataset
+    def models(self):
+        """Models (`~gammapy.modeling.models.Models`)."""
+        return self.grouped._models
+
+    @models.setter
+    def models(self, models):
+        """Models setter"""
+        if self._is_grouped:
```
Comment (registerrier) [inline ogip_spectra/ogip_spectrum_dataset.py:86]: here you probably have to set the models on `self` as well no?
Why `fset` here?

**Review by registerrier — COMMENTED** [review]
Thanks @luca-giunti

File: ogip_spectra/ogip_spectrum_dataset.py (hunk 1 above)
Comment (luca-giunti) [reply ogip_spectra/ogip_spectrum_dataset.py:86]: @registerrier thank you for looking at this.

> here you probably have to set the models on self as well no?

Why is this necessary? I also override the `models` property to redirect to the `self.grouped._models`, so it should be handled correctly internally without the `self` having any model really set on it

 

> Why fset here?

Indeed, no idea. I simiplfied it to `self.grouped.models = models`

A general question: in the future how do we want to integrate this in Gammapy? Add it to the main code (my first choice I guess) or leave it as a user-contributed recipe?

**Comment by luca-giunti** [comment]
@registerrier I addressed your comments and added fitting test. You can see now if it is OK to merge

**Review by registerrier — APPROVED** [review]
Thanks @luca-giunti

**Pull request merged**